src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -157,11 +161,70 @@ protected function addRoute(RouteCollection $collection, $annot, $globals, \Refl
             $condition = $globals['condition'];
         }
 
-        $route = $this->createRoute($globals['path'].$annot->getPath(), $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+        $path = $annot->getPath();
+        $locales = $annot->getLocales();
+        $hasLocalizedPrefix = !empty($globals['locales']);
+        $hasPrefix = $hasLocalizedPrefix || null !== $globals['path'];
+        $isLocalized = !empty($locales);
+        $hasPathOrLocales = null !== $path || $isLocalized;
+
+        if (!$hasPrefix && !$hasPathOrLocales) {
+            throw new \LogicException(sprintf('Annotation for "%s::%s" has no path.', $class->name, $method->name));
+        }
+
+        $paths = array();
+
+        if (!$hasPathOrLocales) {
+            if ($hasLocalizedPrefix) {
+                $paths = $globals['locales'];
+            } else {
+                $paths[] = $globals['path'];
+            }
+        } elseif (!$hasPrefix) {
+            if ($isLocalized) {
+                $paths = $locales;
+            } else {
+                $paths[] = $path;
+            }
+        } elseif ($hasLocalizedPrefix) {
+            if ($isLocalized) {
+                if ($missing = array_diff_key($globals['locales'], $locales)) {
+                    $missingLocales = implode('", "', array_keys($missing));
+                    throw new \LogicException(sprintf('Localized prefix(es) "%s" is/are not defined on "%s" while it is/are defined on its action "%s".', $missingLocales, $class->name, $method->name));
```

Comment:
the other loader has less complex messages in this situation
what about using the simpler version here also? (I don't like the "is/are" :) )